plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java

```diff
@@ -187,7 +188,15 @@ public void setParameters(List<IParameter> parameters) {
 		  parameters.forEach(p -> paramsMap.put(p.getName(), p.getValue()));
 		}
 		originalValueMap = paramsMap;
-		firePropertyChange(PROPERTY_PARAMETERS, this.parameters, this.parameters = injectProjectParameters(this.eclipseProject, parameters));
+		/*
+		 * firePropertyChange will send an event if the array are differents so we need to check
+		 * otherwise we will risk the ObservableList to be based on the wrong list as it is assigned
+		 * in the call.
+		 */
+		List<IParameter> newParameters = injectProjectParameters(this.eclipseProject, parameters);
+		if (!Objects.equals(this.parameters, newParameters)) {
+	        firePropertyChange(PROPERTY_PARAMETERS, this.parameters, this.parameters = newParameters);
```

Comment:
@scabanovich @jeffmaury the pattern that you spotted in several places (actually in most places where the binding is done to a list of items) is required because of a bug in the databinding where it wont work unless you're always using the very same list. I'd fully agree in moving this to a common class. I believe that there are more elegant solution as a util class though.